Sources/TuistGenerator/Mappers/SynthesizedResourceInterfaceProjectMapper.swift

```diff
@@ -161,23 +161,23 @@ public final class SynthesizedResourceInterfaceProjectMapper: ProjectMapping { /
         switch resourceSynthesizer.parser {
         case .strings:
             // This file kind is localizable, let's order files based on it
-            var regionPriorityQueue: [String] = []
-
-            if let developmentRegion {
-                regionPriorityQueue.insert(developmentRegion, at: 0)
-            }
-
-            // Let's sort paths moving the development region localization's one at first
-            let prioritizedPaths = paths.filter { path in
-                regionPriorityQueue.map { path.parentDirectory.basename.contains($0) }.contains(true)
-            }
-
-            let unprioritizedPaths = paths.filter { path in
-                !regionPriorityQueue.map { path.parentDirectory.basename.contains($0) }.contains(true)
-            }
-
-            paths = prioritizedPaths + unprioritizedPaths
-
+            paths = {
+                guard let developmentRegion else { return paths }
+                return paths.sorted { lhs, rhs in
+                    let lhsBasename = lhs.parentDirectory.basenameWithoutExt
+                    let rhsBasename = rhs.parentDirectory.basenameWithoutExt
+
+                    if lhsBasename == developmentRegion {
+                        return true
+                    } else if rhsBasename == developmentRegion {
+                        return false
+                    } else if lhsBasename.contains(developmentRegion), rhsBasename.contains(developmentRegion) {
```

Comment:
If developmentRegion is a substring of another region or language code (e.g., en in zen), this would lead to incorrect prioritisation. (see https://en.wikipedia.org/wiki/List_of_ISO_639-2_codes)

Reply:
That's true, and that bug would already exist in the existing implementation. To fix that, I could make this more explicit by checking whether the `developmentRegion` is a prefix rather than simply a substring, but that still wouldn't give absolute certainty, and there would likely be other edge cases. 

In real terms though, we only need the first file, and so the ordering after an exact match shouldn't really matter.